packages/workflows/src/definitions/store/carts/add-shipping-method/index.ts

```diff
@@ -0,0 +1,303 @@
+import {
+  TransactionStepsDefinition,
+  WorkflowManager,
+} from "@medusajs/orchestration"
+import { CartDTO, WorkflowTypes } from "@medusajs/types"
+import {
+  defaultStoreCartFields,
+  defaultStoreCartRelations,
+} from "@medusajs/utils"
+import {
+  CartHandlers,
+  ShippingMethodHandlers,
+  ShippingOptionHandlers,
+} from "../../../../handlers"
+import { exportWorkflow, pipe } from "../../../../helper"
+import { Workflows } from "../../../../workflows"
+import { prepareAddShippingMethodToCartWorkflowData } from "./prepare-add-shipping-method-to-cart-data"
+import { prepareDeleteShippingMethodsData } from "./prepare-delete-shipping-methods-data"
+import { setRetrieveConfig } from "./set-retrieve-config"
+
+export enum AddShippingMethodWorkflowActions {
+  prepare = "prepare",
+  validateFulfillmentData = "validateFulfillmentData",
+  validateLineItemShipping = "validateLineItemShipping",
+  getOptionPrice = "getOptionPrice",
+  createShippingMethods = "createShippingMethods",
+  cleanUpShippingMethods = "cleanUpShippingMethods",
+  adjustFreeShipping = "adjustFreeShipping",
+  upsertPaymentSessions = "upsertPaymentSessions",
+  result = "result",
+}
+
+export const addShippingMethodWorkflowSteps: TransactionStepsDefinition = {
+  next: [
+    {
+      action: AddShippingMethodWorkflowActions.validateFulfillmentData,
+      noCompensation: true,
+    },
+    {
+      action: AddShippingMethodWorkflowActions.validateLineItemShipping,
+      noCompensation: true,
+      saveResponse: false,
```

Comment:
question: how are these two actions run?

Do they run in parallel? And if so, would the outcome be the same if we moved the tree starting with `getOptionPrice` to the first action?

Reply:
They do run in parallel, but `getOptionPrice` uses the response from `validateFulfillmentData` as an input. Moving it would break the flow.